src/resty/gcp/init.lua

```diff
@@ -51,50 +51,116 @@ FindApis = function(apiClass, methods, curr)
     return methods
 end
 
+local function template_expansion(str, params)
+    return (string.gsub(str, "{(.-)}", function (key)
+        return ngx.escape_uri(params[key])
+    end))
+end
+
+local function build_request(accesstoken, apiDetail, baseUrl, params, requestBody, opt)
+    local media = opt and opt.media
+    if media and not apiDetail.supportsMediaDownload then
+        error("API does not supported media download")
+    end
+
+    local mediaUpload = apiDetail.supportsMediaUpload
```

Comment:
Would you mind adding a (busted) unit test or two for this `build_request()` function to your change set?

Right now our test coverage for this library is pretty much nil. I suspect that much of this request-handling code will be refactored and potentially rewritten as we try to make the library more mature, and it would be good to have _some_ concrete test cases to work with when that effort happens.

Reply:
done